Marlin/src/module/probe.h

```diff
@@ -152,6 +152,9 @@ class Probe {
         #else
           _MAX((X_MIN_BED) + (PROBING_MARGIN_LEFT), (X_MIN_POS) + offset_xy.x)
         #endif
+        #if ENABLED(NOZZLE_AS_PROBE)
+          - home_offset[X_AXIS]
+        #endif
```

Comment:
@InsanityAutomation — Do you remember why you only wanted to apply the `home_offset` to probe movement limits when `NOZZLE_AS_PROBE` was enabled? The PR #20617 proposes to change this so `home_offset` is applied for all probes, regardless of whether they have offsets or not.

Reply:
IIRC from our discussion way back then, you wanted to keep the offsets respecting workspaces applied in G29 and hesitated to apply it here, so we only applied it for the one that could not be edited by gcode. I personally have no aversion to the home offset being applied globally.